service/entityresolution/keycloak/keycloak_entity_resolution.go

```diff
@@ -28,12 +30,18 @@ const UsernameConditionalSelector = "client_id"
 const serviceAccountUsernamePrefix = "service-account-"
 
 type KeycloakConfig struct {
-	URL            string `json:"url"`
-	Realm          string `json:"realm"`
-	ClientID       string `json:"clientid"`
-	ClientSecret   string `json:"clientsecret"`
-	LegacyKeycloak bool   `json:"legacykeycloak" default:"false"`
-	SubGroups      bool   `json:"subgroups" default:"false"`
+	URL                 string              `json:"url"`
+	Realm               string              `json:"realm"`
+	ClientID            string              `json:"clientid"`
+	ClientSecret        string              `json:"clientsecret"`
+	LegacyKeycloak      bool                `json:"legacykeycloak" default:"false"`
+	SubGroups           bool                `json:"subgroups" default:"false"`
+	UnknownEntityConfig UnknownEntityConfig `json:"unknownentities" default:"false"`
+}
+
+type UnknownEntityConfig struct {
+	Enabled     bool     `json:"enabled"`
+	EntityTypes []string `json:"entitytypes"`
```

Comment:
I'd rather not have bad configs cause failures during runtime; instead maybe do a custom type here so we can validate this during parse?

https://sagikazarmark.hu/blog/decoding-custom-formats-with-viper/

Reply:
In another repo we use https://github.com/go-playground/validator to validate the config at runtime and indicate where the error occurs. Not sure if that would be useful here.

Reply:
can i break this into a separate issue? cause we have the same problem with the authorization config parsing

Reply:
actually i see sean did something with the validator for the authz config in his opa pr, i'll just dupe that here

Reply:
I simplified the type so it no longer contains string comparisons at runtime